src/sys/unix/net.rs

```diff
@@ -29,7 +29,8 @@ pub(crate) fn new_socket(
         target_os = "freebsd",
         target_os = "linux",
         target_os = "netbsd",
-        target_os = "openbsd"
+        target_os = "openbsd",
+        target_os = "illumos"
```

Comment:
Nit: can you sort these alphabetically, same for all places you've added `illumos` target.

Reply:
Yep.  I'll fix this one and the others.